src/main/scala/dpla/ingestion3/mappers/providers/BhlMapping.scala

```diff
@@ -0,0 +1,230 @@
+  // <mods:relatedItem @type="series"><titleInfo><title>
+    (data \\ "metadata" \ "mods" \ "relatedItem")
+      .flatMap(n => getByAttribute(n.asInstanceOf[Elem], "type", "series"))
+      .flatMap(n => extractStrings(n \ "titleInfo" \ "title"))
+      .map(nameOnlyCollection)
+
+  override def contributor(data: Document[NodeSeq]): ZeroToMany[EdmAgent] = {
+    // <mods:name><mods:namePart> when any of <mods:name><mods:role><mods:roleTerm> equals contributor
+    // concatenate <mods:namePart> when @type does not equal "affiliation", "displayForm", "description", or "role"
+
+    val badTypes = List("affiliation", "displayForm", "description", "role")
+
+    val parentNodes: NodeSeq = (data \\ "metadata" \ "mods" \ "name")
+      .flatMap(_.filter(n => (n \\ "roleTerm").map(_.text.toLowerCase).contains("contributor")))
+
+    val names: Seq[String] = parentNodes.map(n =>
+      (n \ "namePart")
+        .filterNot(n => badTypes.contains(n \@ "type"))
+        .flatMap(extractStrings)
+        .map(_.cleanupEndingCommaAndSpace)
+        .mkString(", ")
+    )
+
+    names.map(nameOnlyAgent)
+  }
+
+  override def creator(data: Document[NodeSeq]): ZeroToMany[EdmAgent] = {
+    // <mods:name><mods:namePart> when any of <mods:name><mods:role><mods:roleTerm> equals creator
+    // concatenate <mods:namePart> when @type does not equal "affiliation", "displayForm", "description", or "role"
+
+    val badTypes = List("affiliation", "displayForm", "description", "role")
+
+    val parentNodes: NodeSeq = (data \\ "metadata" \ "mods" \ "name")
+      .flatMap(_.filter(n => (n \\ "roleTerm").map(_.text.toLowerCase).contains("creator")))
+
+    val names: Seq[String] = parentNodes.map(n =>
+      (n \ "namePart")
+        .filterNot(n => badTypes.contains(n \@ "type"))
+        .flatMap(extractStrings)
+        .map(_.cleanupEndingCommaAndSpace)
+        .mkString(", ")
+    )
+
+    names.map(nameOnlyAgent)
+  }
+
+  override def date(data: Document[NodeSeq]): ZeroToMany[EdmTimeSpan] = {
+    // <mods:originInfo><mods:dateIssued keyDate="yes"> OR
+    // <mods:originInfo><mods:dateOther type="issueDate" keyDate="yes>
+    // Records should have only one or the other
+
+    val dateIssued: Seq[String] = (data \\ "metadata" \ "mods" \ "originInfo" \ "dateIssued")
+      .flatMap(n => getByAttribute(n.asInstanceOf[Elem], "keyDate", "yes"))
+      .flatMap(extractStrings)
+
+    val dateOther: Seq[String] = (data \\ "metadata" \ "mods" \ "originInfo" \ "dateOther")
+      .flatMap(n => getByAttribute(n.asInstanceOf[Elem], "keyDate", "yes"))
+      .flatMap(n => getByAttribute(n.asInstanceOf[Elem], "type", "issueDate"))
+      .flatMap(extractStrings)
+
+    getDateValues(dateIssued ++ dateOther)
+      .map(stringOnlyTimeSpan)
+  }
+
+  def getDateValues(dates: Seq[String]): Seq[String] = {
+    val sorted = dates.sorted
+
+    sorted.reverse.headOption match { // get last date in sequence
```

Comment:
The dates are sorted here b/c they are sorted in Ingestion 1 (see https://github.com/dpla/ingestion/blob/develop/lib/mappers/bhl_mods.py#L300 )  As far as I can tell, its so that in a situation like this:
```
<dateIssued>2000</dateIssued>
<dateIssued>1980</dateIssued>
```
you will be sure that when you concatenate them to for a date range, you get `1980-2000` and not `2000-1980`.

I also think that its to force date ranges after individual dates, so that in a situation like this:
```
<dateIssued>1980-2000</dateIssued>
<dateIssued>1980</dateIssued>
```
you will be sure that `1980-2000` will be at the end of the sorted list, and you can pop it off and ignore `1980`.

It seems like maybe not the best logic, but I was just trying to duplicate what was in ingest 1. 

Also thanks for that `lastOption` tip :)